vimeo/vimeo-networking-java
# VA-1367 Stag integration for custom type adapters

#### Ticket
[VA-1367](https://vimean.atlassian.net/browse/VA-1367)

#### Ticket Summary
Add custom type adapters

#### Implementation Summary
Using the new stag library, adding support for all model objects, except lists. Also, this doesn't support enums yet. For both BaseResponseList types and enums, the default Gson Type Adapter is used (EnumTypeAdapter and Reflective). I created a [ticket](https://vimean.atlassian.net/browse/VA-1793) to update those as well.
Currently, performance on the Nexus 9 is the same as pre-stag, which is just slightly higher than without custom type adapters (by the same percentage higher). I have not tested on the S2 yet, as I am WFH while submitting this PR, and only took the Nexus 9, but I will do profiling on the S2 as well.

#### How to Test
Verify the library compiles for you (will have to init/update the submodule) and the app runs as normal.
